project/src/test/java/com/google/sps/SearchDataServletTest.java

```diff
@@ -0,0 +1,146 @@
+  private HttpServletRequest postRequest;
+  private HttpServletResponse postResponse;
+  private HttpServletRequest secondPostRequest;
+  private HttpServletResponse secondPostResponse;
+  private HttpServletRequest getRequest;
+  private HttpServletResponse getResponse;
+  private StringWriter stringWriter;
+  private PrintWriter printWriter;
+  private SearchDataServlet searchDataServlet;
+  private static final String EVENT_ID_1 = TestUtils.newRandomId();
+  private static final String EVENT_ID_2 = TestUtils.newRandomId();
+  private static final String PARAMETER_KEYWORD = "keyword";
+  private static final String PARAMETER_EVENT_ID = "event-id";
+  private static final String PARAMETER_DESCRIPTION = "description";
+  private static final String WALKING = "walking";
+  private static final String CAKE = "cake";
+  private static final String MUSIC = "music";
+ 
+  @Before
+  public void setUp() throws Exception {
+    postRequest = Mockito.mock(HttpServletRequest.class);
+    postResponse = Mockito.mock(HttpServletResponse.class);
+
+    secondPostRequest = Mockito.mock(HttpServletRequest.class);
+    secondPostResponse = Mockito.mock(HttpServletResponse.class);
+
+    getRequest = Mockito.mock(HttpServletRequest.class);
+    getResponse = Mockito.mock(HttpServletResponse.class);
+
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+    Mockito.when(getResponse.getWriter()).thenReturn(printWriter);
+
+    searchDataServlet = new SearchDataServlet();
+  }
+
+  @Test
+  public void addEventAndRetrieveResultsForKeywordNotInEventDescription_noResultsReturned()
+      throws IOException {
+    Mockito.when(postRequest.getParameter(PARAMETER_DESCRIPTION))
+        .thenReturn(newDescriptionWithoutCakeKeyword());
+    Mockito.when(postRequest.getParameter(PARAMETER_EVENT_ID)).thenReturn(EVENT_ID_1);
+    Mockito.when(getRequest.getParameter(PARAMETER_KEYWORD)).thenReturn(CAKE);
+
+    searchDataServlet.doPost(postRequest, postResponse);
+    searchDataServlet.doGet(getRequest, getResponse);
+
+    Assert.assertEquals(CommonUtils.convertToJson(Arrays.asList()), stringWriter.toString().trim());
+  }
+
+  @Test
+  public void addEventAndRetrieveResultsForKeywordInEventDescription_oneResultReturned()
+      throws IOException {
+    Mockito.when(postRequest.getParameter(PARAMETER_DESCRIPTION))
+        .thenReturn(newDescriptionWithWalkingKeyword());
+    Mockito.when(postRequest.getParameter(PARAMETER_EVENT_ID)).thenReturn(EVENT_ID_1);
+    Mockito.when(getRequest.getParameter(PARAMETER_KEYWORD)).thenReturn(WALKING);
+
+    searchDataServlet.doPost(postRequest, postResponse);
+    searchDataServlet.doGet(getRequest, getResponse);
+
+    Assert.assertEquals(
+        CommonUtils.convertToJson(Arrays.asList(EVENT_ID_1)).trim(),
+        stringWriter.toString().trim());
+  }
+
+  @Test
+  public void addTwoEventsAndRetrieveResultsForKeywordInDescriptionOfTwoEvents_twoResultsReturned()
+      throws IOException {
```

Comment:
So basically when there is a search with N words in the production website, there will be N post requests called? Why do you not call once?

Reply:
Here, post is supposed to happen on insertion of an event but I do not want to touch the EventCreation servlet before we have tested our MVP.